### ISSUE TITLE
Plugin seems to ignore (not apply) .snyk exclusion file

### ISSUE THREAD

**Issue by marconew10** [snyk/snyk-maven-plugin#29]
- `mvn --version`: 
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T20:41:47+02:00)
Java version: 1.8.0_202, vendor: AdoptOpenJdk, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre
Default locale: en_NL, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.4", arch: "x86_64", family: "mac"

### Expected behaviour
When running _mvn snyk:test_ a High vulnerability is found.   That vulnerability I have entered in a _.snyk_ file in the root of the project:

I expect _mvn snyk:test_ to **not** report this vulnerability anymore and the given maven command to succeed with BUILD SUCCESSFUL.  When running the (non-maven) command _snyk test_ indeed the vulnerability is not reported (detected, but skipped for reporting/failure).

### Actual behaviour
The _mvn snyk:test_ command keeps reporting the High vulnerability, while it is listed in the _.snyk_ file.  The maven command fails with BUILD FAILURE.   Also tried putting the _.snyk_ file in subdirectories, but that didn't work either.

### Steps to reproduce
Output of _mvn snyk:test_ without the .snyk file, so expecting to fail:
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< com.ignoring:dot-snyk-file >---------------------
[INFO] Building Sample project showing .snyk is ignored 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- snyk-maven-plugin:1.1.1:test (default-cli) @ dot-snyk-file ---
[WARNING] ✗ medium severity vulnerability found on com.google.guava:guava@18.0
[WARNING] - desc: Deserialization of Untrusted Data
[WARNING] - info: https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236
[WARNING] - from: com.ignoring:dot-snyk-file@1.0-SNAPSHOT > com.amazonaws:amazon-kinesis-producer@0.12.11 > com.google.guava:guava@18.0
[WARNING] 
[WARNING] ✗ high severity vulnerability found on com.google.protobuf:protobuf-java@2.6.1
[WARNING] - desc: Integer Overflow
[WARNING] - info: https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-173761
[WARNING] - from: com.ignoring:dot-snyk-file@1.0-SNAPSHOT > com.amazonaws:amazon-kinesis-producer@0.12.11 > com.google.protobuf:protobuf-java@2.6.1
[WARNING] 
[WARNING] Tested 17 dependencies for known vulnerabilities, found 2 vulnerabilities, 2 vulnerable paths.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.815 s
[INFO] Finished at: 2019-06-17T12:31:19+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.snyk:snyk-maven-plugin:1.1.1:test (default-cli) on project dot-snyk-file: Found vulnerabilities with severity high. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

Now added this .snyk file:
```
version: v1.5.0
ignore:
  'SNYK-JAVA-COMGOOGLEPROTOBUF-173761':
    - '* > com.google.protobuf:protobuf-java':
        reason: 'No newer version of KCL and KPL available'
        expires: '2020-02-15T01:00:00.000Z'
```

Output of _mvn snyk:test_ after adding the above .snyk file in the root level of the project, at same level of pom.xml. So not expecting the build to fail anymore:
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< com.ignoring:dot-snyk-file >---------------------
[INFO] Building Sample project showing .snyk is ignored 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- snyk-maven-plugin:1.1.1:test (default-cli) @ dot-snyk-file ---
[WARNING] ✗ medium severity vulnerability found on com.google.guava:guava@18.0
[WARNING] - desc: Deserialization of Untrusted Data
[WARNING] - info: https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236
[WARNING] - from: com.ignoring:dot-snyk-file@1.0-SNAPSHOT > com.amazonaws:amazon-kinesis-producer@0.12.11 > com.google.guava:guava@18.0
[WARNING] 
[WARNING] ✗ high severity vulnerability found on com.google.protobuf:protobuf-java@2.6.1
[WARNING] - desc: Integer Overflow
[WARNING] - info: https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-173761
[WARNING] - from: com.ignoring:dot-snyk-file@1.0-SNAPSHOT > com.amazonaws:amazon-kinesis-producer@0.12.11 > com.google.protobuf:protobuf-java@2.6.1
[WARNING] 
[WARNING] Tested 17 dependencies for known vulnerabilities, found 2 vulnerabilities, 2 vulnerable paths.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.077 s
[INFO] Finished at: 2019-06-17T12:33:42+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.snyk:snyk-maven-plugin:1.1.1:test (default-cli) on project dot-snyk-file: Found vulnerabilities with severity high. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

Maven pom.xml used:
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>com.ignoring</groupId>
  <artifactId>dot-snyk-file</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>Sample project showing .snyk is ignored</name>

  <properties>
    <java.version>1.8</java.version>
    <resource.delimiter>@</resource.delimiter> <!-- delimiter that doesn't clash with Spring ${} placeholders -->
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
    <maven.compiler.source>${java.version}</maven.compiler.source>
    <maven.compiler.target>${java.version}</maven.compiler.target>

    <skipTests>false</skipTests>
    <skipITs>${skipTests}</skipITs>
    <skipUTs>${skipTests}</skipUTs>  <!-- Allows to run only integration tests without surefire unit tests specifying '-DskipUTs' -->
  </properties>

  <dependencies>

    <dependency>
      <groupId>com.amazonaws</groupId>
      <artifactId>amazon-kinesis-producer</artifactId>
      <version>0.12.11</version>
    </dependency>

    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>2.9.9</version>
    </dependency>

  </dependencies>

  <build>
    <pluginManagement>
      <plugins>
        <plugin>
          <groupId>io.snyk</groupId>
          <artifactId>snyk-maven-plugin</artifactId>
          <version>1.1.1</version>
          <executions>
            <execution>
              <id>snyk-test</id>
              <phase>test</phase>
              <goals>
                <goal>test</goal>
              </goals>
            </execution>
            <execution>
              <id>snyk-monitor</id>
              <phase>install</phase>
              <goals>
                <goal>monitor</goal>
              </goals>
            </execution>
          </executions>
          <configuration>
            <apiToken>${env.SNYK_TOKEN}</apiToken>
            <failOnSeverity>high</failOnSeverity>
            <org>new10</org>
          </configuration>
        </plugin>
      </plugins>
    </pluginManagement>
  </build>

</project>
```

I could attach whole testproject if needed.

Best regards,
Marco

**Comment by pavel-github** [comment]
@marconew10, the fix is already submitted with commit 6f1f10c570357f50112af02cd6d1c223ddd24795.
So please update plugin version to 2.0 or above.

**Comment by marconew10** [comment]
Hello Pavel,


A nice. But I can't find any 2.0 version or higher. I only see in 'master' branch version 1.2.1? https://github.com/snyk/snyk-maven-plugin/blob/master/pom.xml

Or am I missing something?


Best regards,

Marco

New10


________________________________

**Comment by pavel-github** [comment]
@marconew10, shame on me :) typo. it should be 1.2.0 or higher. `1.2.1` is perfect.

**Comment by pavel-github** [comment]
Hello Marco,
any luck with the new plugin version `1.2.1`?
Best regards,
Pavel

**Comment by marconew10** [comment]
Hi Pavel,


Yes it seems to do its thing correctly, so issue can be closed (thought it was already :).


Best regards,

Marco

New10

________________________________

**Comment by pavel-github** [comment]
👍 thanks for update.

**Comment by dnsbtchr** [comment]
Sorry to bring this up again but I can't get the plugin to ignore a found vulnerability using the `.snyk` file. 
My plugin looks like this:
`      <plugin>
        <groupId>io.snyk</groupId>
        <artifactId>snyk-maven-plugin</artifactId>
        <version>1.2.6</version>
        <executions>
          <execution>
            <id>snyk-test</id>
            <phase>verify</phase>
            <goals>
              <goal>test</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <apiToken>xxx</apiToken>
          <failOnSeverity>medium</failOnSeverity>
          <org>xxx</org>
        </configuration>
      </plugin>`
And my `.snyk` file in the project root looks like this:
`version: v1.5.0
ignore:
  'SNYK-JAVA-ORGYAML-537645':
    - '* > org.codehaus.jackson:jackson-mapper-asl':
        reason: 'My reason'
        expires: '2021-01-01T00:00:00.000Z'
`
It's probably just a stupid thing I miss but I'd be super grateful if you can help me.
Thanks

**Comment by marconew10** [comment]
@dnsbtchr 

I have the plugin in phase 'test' (you have 'verify'). Should not matter normally I guess. I'm using 1.2.1 as version, maybe give that a try?
And the formatting of the .snyk file is really important. Two spaces indentation each time. My .snyk file in my first message in this thread has the right formatting, maybe double check?

I also have the monitor set up in the plugin, might affect it:
 ```
<execution>
              <id>snyk-monitor</id>
              <phase>install</phase>
              <goals>
                <goal>monitor</goal>
              </goals>
            </execution>

```

Snyk file has to be in the root of the project.

I'm always also running the monitor command:
```
mvn snyk:monitor
mvn snyk:test
```

PS: for formatting, use the <> icon in the top menu above each post :)

Hope this helps!

**Comment by dnsbtchr** [comment]
Hey @marconew10, thanks for the quick answer. Unfortunately it didn't help. 
I tried:
- Changing the phase to validate and test
- Using version 1.2.1
- Adding the monitor goal
- Copy & paste of your .snyk file content and replacing the vulnerability ID

Nothing worked. It feels like it simply ignores my .snyk file. I ran it with -X to see the debug output but unfortunately I couldn't find anything about the file.
